### ISSUE TITLE
Var Sources for Credential Management

### ISSUE THREAD

**Issue by matthewpereira** [concourse/concourse#5663]
### TLDR

* `var_sources` are a way to configure multiple named credential managers, initially at the pipeline-level but potentially at the system-level and project-level (concourse/rfcs#32).
* A var source prototype is an interface implemented by a prototype which allows it to be used as a var source in a pipeline.

### Problem

- Credential management is a hot topic for engineers and teams operating Concourse installations. 

- Concourse currently supports configuring a single credential manager for the entire cluster. This is limiting in a number of ways, especially for enterprise organizations that need to scale Concourse usage across business lines and multiple-team environments.

### Proposed Solution

- This proposal formalizes the `((var))` convention and upgrades it with additional context. `var_sources` can be configured at the pipeline level, and `((var))`s can now be expressed as `((VAR_SOURCE_NAME:SECRET_PATH.SECRET_FIELD))` to point to the correct context.

See the RFC for more information: https://github.com/concourse/rfcs/pull/39

### Outcomes

- A Concourse pipeline can now support multiple named credential managers.

### Current Status

- In Progress

### References

#5229 - Extensible var sources
RFC: https://github.com/concourse/rfcs/pull/39
